python/ray/_raylet.pyx

```diff
@@ -595,12 +596,24 @@ cdef CRayStatus task_execution_handler(
                     "worker_crash",
                     traceback_str,
                     job_id=None)
-                sys.exit(1)
+                # Cython's bug that doesn't allow reference assignment
```

Comment:
When I create a faulty actor now, I see a super ugly stack trace:
```
>>> A.remote()
Actor(A,480a853c2c4c6f276de42d0101000000)
>>> 2021-02-19 18:34:55,626	WARNING worker.py:1099 -- Traceback (most recent call last):
  File "python/ray/_raylet.pyx", line 587, in ray._raylet.task_execution_handler
  File "python/ray/_raylet.pyx", line 445, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 554, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 482, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 485, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 495, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 443, in ray._raylet.execute_task.function_executor
  File "/home/eric/Desktop/ray/python/ray/function_manager.py", line 556, in actor_method_executor
    return method(__ray_actor, *args, **kwargs)
  File "<stdin>", line 4, in __init__
AssertionError
An unexpected internal error occurred while the worker was executing a task.
(pid=19620) 2021-02-19 18:34:55,619	ERROR worker.py:378 -- Error raised in creation task
(pid=19620) Traceback (most recent call last):
(pid=19620)   File "python/ray/_raylet.pyx", line 587, in ray._raylet.task_execution_handler
(pid=19620)   File "python/ray/_raylet.pyx", line 445, in ray._raylet.execute_task
(pid=19620)   File "python/ray/_raylet.pyx", line 554, in ray._raylet.execute_task
(pid=19620)   File "python/ray/_raylet.pyx", line 482, in ray._raylet.execute_task
(pid=19620)   File "python/ray/_raylet.pyx", line 485, in ray._raylet.execute_task
(pid=19620)   File "python/ray/_raylet.pyx", line 495, in ray._raylet.execute_task
(pid=19620)   File "python/ray/_raylet.pyx", line 443, in ray._raylet.execute_task.function_executor
(pid=19620)   File "/home/eric/Desktop/ray/python/ray/function_manager.py", line 556, in actor_method_executor
(pid=19620)     return method(__ray_actor, *args, **kwargs)
(pid=19620)   File "<stdin>", line 4, in __init__
(pid=19620) AssertionError
(pid=19620) 
(pid=19620) The above exception was the direct cause of the following exception:
(pid=19620) 
(pid=19620) Traceback (most recent call last):
(pid=19620)   File "python/ray/_raylet.pyx", line 608, in ray._raylet.task_execution_handler
(pid=19620) SystemExit: 1
2021-02-19 18:35:00,628	ERROR worker.py:1045 -- Possible unhandled error from worker: ray::A.__init__() (pid=19620, ip=192.168.5.121)
  File "python/ray/_raylet.pyx", line 495, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 443, in ray._raylet.execute_task.function_executor
  File "<stdin>", line 4, in __init__
AssertionError
```

In current Ray, nothing is printed if an exception is failing. Can you fix the PR so the behavior is preserved?